weave

```diff
@@ -1393,51 +1425,17 @@ launch_router() {
         --http-addr $HTTP_ADDR \
         --status-addr $STATUS_ADDR \
         --resolv-conf "/var/run/weave/etc/$RESOLV_CONF_BASE" \
-        "$@")
+        $PROXY_ARGS)
     wait_for_status $CONTAINER_NAME http_call $HTTP_ADDR
+    # Wait for Weave Proxy:
+    [ -n "$NO_PROXY" ] || wait_for_status $CONTAINER_NAME http_call_unix $CONTAINER_NAME status.sock
```

Comment:
I mean "it shouldn't be necessary because `weaver` should ensure that everything is up and running before answering the main status call".  But I see that isn't actually the case - it calls `ListenAndServeHTTP` before starting the plugin and the proxy.  In the plugin case this is perhaps to make the `EnsureNetwork` call work.  

I can't see a reason to start the proxy after the main http listener.